samza-core/src/main/java/org/apache/samza/config/JavaSystemConfig.java

```diff
@@ -122,4 +125,27 @@ public SystemAdmin getSystemAdmin(String systemName) {
   public Config getDefaultStreamProperties(String systemName) {
     return subset(String.format(SYSTEM_DEFAULT_STREAMS_PREFIX_FORMAT, systemName), true);
   }
+
+  /**
+   * Get SYSTEM reset offset value.
+   * Check two places - system default and stream system default.
+   * Stream default system is the right way to set it.
+   * systems.system.samza.offset.defautl is the deprecated setting, but needs to be checked for backward compatibility
+   * @param systemName get config value for this system
+   * @return value of system reset or default ("upcoming") if none set
+   */
+  public String getSystemOffsetDefault(String systemName) {
+    // first check stream system default
```

Comment:
Remove references to "stream system default" and "stream default system" here and in javadoc. Replace with "system default stream offset default". It's verbose, but at least consistent and correct.